### PR TITLE
`Topic` extensions for Message Router client

### PR DESCRIPTION
**Pull request by BalassaMarton**
Added the `Topic` extension method for obtaining an `IAsyncObservable` for a given topic.

### CODE REVIEW

File: src/messaging/dotnet/src/Core/Internal/Disposable.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        public void Dispose()
+        {
+            var task = _asyncDisposable.DisposeAsync();
+
+            if (task.IsCompletedSuccessfully)
+                return;
+
+            task.GetAwaiter().GetResult();
```
Comment (BalassaMarton) [inline src/messaging/dotnet/src/Core/Internal/Disposable.cs:46]: Up for debate. Alternative is to fire and forget - it doesn't have much impact, the subscription is completed synchronously.

File: src/messaging/dotnet/src/Core/Internal/Disposable.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+internal static class Disposable
+{
+    public static IDisposable FromAsyncDisposable(IAsyncDisposable asyncDisposable) =>
+        new AsyncDisposableWrapper(asyncDisposable);
+
+    public static ValueTask<IDisposable> FromAsyncDisposable(ValueTask<IAsyncDisposable> awaitable)
+    {
+        return awaitable.IsCompletedSuccessfully 
+            ? new ValueTask<IDisposable>(FromAsyncDisposable(awaitable.Result)) 
+            : FromAsyncDisposableImpl(awaitable);
```
[the remark this answers is not in the document]
Comment (BalassaMarton) [reply src/messaging/dotnet/src/Core/Internal/Disposable.cs:24]: How do you mean? This is a pretty standard way of returning a `ValueTask` and it's about twice as fast on the happy path when `awaitable` is already completed (just measured).

[the remark this answers is not in the document]
Comment (ZKRobi) [reply src/messaging/dotnet/src/Core/Internal/Disposable.cs:24]: FromAsyncDisposable calls FromAsyncDisposableImpl calls another overload of FromAsyncDisposable caused a wtf moment before I understood what's going on. It was a bit hard to read for me, that's all.

File: src/messaging/dotnet/src/Core/Internal/Disposable.cs (hunk 1 above)
Comment (ZKRobi) [reply src/messaging/dotnet/src/Core/Internal/Disposable.cs:46]: I agree with the current solution

**Review by ZKRobi — APPROVED** [review]

**Pull request merged**